javatests/arcs/android/systemhealth/testapp/StorageCore.kt

```diff
@@ -378,6 +379,17 @@ class StorageCore(val context: Context, val lifecycle: Lifecycle) {
                 }
             }
 
+            // The very first SingletonHandle is responsible for writing an entity
+            // to storage then creating its reference.
+            if (taskId == 0) {
+                withContext(handle.dispatcher) {
+                    handle.store(SystemHealthTestEntity.referencedEntity)
+                }.join()
+
+                SystemHealthTestEntity.entityReference = handle.createReference(
+                    SystemHealthTestEntity.referencedEntity
+                ).toReferencable()
```

Comment:
This could be:

```suggestion
                SystemHealthTestEntity.entityReference = withContext(handle.dispatcher) {
                    handle.store(SystemHealthTestEntity.referencedEntity).join()
                    handle.createReference(SystemHealthTestEntity.referencedEntity).toReferencable()
                }
```

Reply:
I remembered you suggested not join() in scheduler dispatcher.
Is it still true now?

Reply:
It should be fine now... once the `join` suspends, it'll queue another task onto the Scheduler Dispatcher

Reply:
gotcha, thanks